### DISCUSSION TITLE
Background variant not working

### DISCUSSION THREAD

**Discussion by moyalmic** [bcakmakoglu/vue-flow#1297] [Q&A] [answered]
I've been trying to get the background variant to work in my project, but it doesn't seem to respect the enum i'm sending in. Bellow is the screenshot of my code, the bgColor does actually take effect and changes correctly, but the Lines (or dots for that matter) don't appear. Am I missing something?
![image](https://github.com/bcakmakoglu/vue-flow/assets/98823105/d46c70c2-651a-4739-8356-ef14bf5b2005)

**Comment by bcakmakoglu** [comment, marked as answer]
You can't use both a bg color and a pattern color on the Background component.
the bg color prop is sort of useless since you can just use `<VueFlow :style="{ backgroundColor: color }">`

**Reply by bcakmakoglu** [reply]
`bgColor` & `patternColor` will be deprecated with the next minor release of `@vue-flow/background`.

There will be a replacement prop called `color` which applies to the *pattern*.

To set a *background color* you can use, as I suggested above, css instead :)

**Comment by moyalmic** [comment]
The problem wasn't with the color though, it was that the variant isn't being respected and neither a lines or dots grid is being rendered...

**Reply by bcakmakoglu** [reply]
The problem *is* with the color.
If you use a bg-color *there will be no pattern*.
It is *either or* not *both*.

**Reply by bcakmakoglu** [reply]
Here are some example to make it clearer:
```vue
<!-- This will ONLY show a background color, NO pattern; even if you use `pattern-color`, `bg-color` will OVERWRITE it -->
<Background pattern-color="#fefefe" bg-color="#000" />
<Background bg-color="#000" />

<!-- This will show a dots pattern (default) - both work the same -->
<Background pattern-color="#fefefe" />
<Background variant="dots" pattern-color="#fefefe" />

<!-- This will show a lines pattern -->
<Background variant="lines" pattern-color="#fefefe" />

**Reply by moyalmic** [reply]
Oh, I understand now. Thanks for the clarification. 

So how would I achieve a custom background color with custom pattern colors? I suppose I could color the div containing the VewFlow component?

**Reply by bcakmakoglu** [reply]
You can just add a background color to the `VueFlow` component :)

```vue
<VueFlow style="background-color: hotpink">...
```
